src/eytzinger_benchmark.zig

```diff
@@ -0,0 +1,319 @@
+    {
+        var benchmark = try Benchmark.begin();
+        var i: usize = 0;
+        var v: usize = 0;
+        while (i < layout.searches) : (i += 1) {
+            const page_index = page_picker[i % page_picker.len];
+            const target = value_picker[v % value_picker.len];
+            const page = pages[page_index];
+            const bounds = Eytzinger.search_values(Key, Val, Val.key_compare, &page.summary, &page.values, target);
+            assert(bounds.len != 0);
+            const hit = if (bounds.len == 1) bounds[0]
+                else bounds[binary_search(Key, Val, Val.key_from_value, Val.key_compare, bounds, target)];
+
+            assert(hit.key == target);
+            if (i % pages.len == 0) v += 1;
+        }
+
+        const result = try benchmark.end(layout.searches);
+        try stdout.print(body_fmt, .{
+            layout.key_size,
+            layout.value_size,
+            layout.keys_count,
+            layout.values_count,
+            "E",
+            "B",
+            result.wall_time,
+            result.utime,
+            result.cpu_cycles,
+            result.instructions,
+            result.cache_references,
+            result.cache_misses,
+            result.branch_misses,
+        });
+    }
+
+    {
+        var benchmark = try Benchmark.begin();
+        var i: usize = 0;
+        var v: usize = 0;
+        while (i < layout.searches) : (i += 1) {
+            const target = value_picker[v % value_picker.len];
+            const p = page_picker[i % page_picker.len];
+            const page = pages[p];
+            const hit = page.values[binary_search(Key, Val, Val.key_from_value, Val.key_compare, page.values[0..], target)];
+
+            assert(hit.key == target);
+            if (i % pages.len == 0) v += 1;
+        }
+        const result = try benchmark.end(layout.searches);
+        try stdout.print(body_fmt, .{
+            layout.key_size,
+            layout.value_size,
+            layout.keys_count,
+            layout.values_count,
+            "_",
+            "B",
+            result.wall_time,
+            result.utime,
+            result.cpu_cycles,
+            result.instructions,
+            result.cache_references,
+            result.cache_misses,
+            result.branch_misses,
+        });
+    }
+}
+
+const Layout = struct {
+    blob_size: usize, // bytes allocated for all pages
+    key_size: usize, // bytes per key
+    value_size: usize, // bytes per value
+    keys_count: usize, // keys per page (in the summary)
+    values_count: usize, // values per page
+    searches: usize,
+};
+
+fn Value(comptime layout: Layout) type {
+    return struct {
+        pub const max_key = 1 << (8 * layout.key_size) - 1;
+        pub const Key = math.IntFittingRange(0, max_key);
+        const Self = @This();
+        key: Key,
+        body: [layout.value_size - layout.key_size]u8,
+
+        comptime {
+            assert(@sizeOf(Key) == layout.key_size);
```

Comment:
We should also assert that the size of the `Value` overall is exactly `value_size` plus `key_size`. Otherwise, things might not be cache-line aligned.